packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-11/wallet-all-ertp/gen-game-offer.mjs

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable @jessie.js/safe-await-separator */
+/* global process */
+import assert from 'assert';
+import { execFile } from 'child_process';
+
+const ISTunit = 1_000_000n; // aka displayInfo: { decimalPlaces: 6 }
+const CENT = ISTunit / 100n;
+
+const [_node, _script, ...choices] = process.argv;
+
+const toSec = ms => BigInt(Math.round(ms / 1000));
+
+const id = `join-${Date.now()}`;
+
+// look up boardIDs for brands, instances in agoricNames
+
+// poor-man's zx
+const $ = cmd => {
+  console.error(cmd);
+  const [file, ...args] = cmd.split(' ');
+
+  return new Promise((resolve, reject) => {
+    execFile(file, args, { encoding: 'utf8' }, (err, out) => {
+      if (err) return reject(err);
+      resolve(out);
+    });
+  });
+};
+
+const zip = (xs, ys) => xs.map((x, i) => [x, ys[i]]);
+const fromSmallCapsEntries = txt => {
+  const { body, slots } = JSON.parse(txt);
+  const theEntries = zip(JSON.parse(body.slice(1)), slots).map(
+    ([[name, ref], boardID]) => {
+      const iface = ref.replace(/^\$\d+\./, '');
```

Comment:
this has got poor-man's-smallcaps-decoding mixed in. The smallcaps-related hacks could at least be moved into the `const smallCaps = {...}` widget.